core/src/test/scala/unit/kafka/cluster/PartitionTest.scala

```diff
@@ -3167,7 +3307,8 @@ class PartitionTest extends AbstractPartitionTest {
     replicaId: Int,
     lastCaughtUpTimeMs: Long,
     logEndOffset: Long,
-    logStartOffset: Long
+    logStartOffset: Long,
+    brokerEpoch: Long = -2
```

Comment:
Could we define a constant for -2 and give it a meaningful name?

Reply:
Refactor to Option[long]. Using null to indicate that we need to verify the broker epoch.